torch/fx/experimental/proxy_tensor.py

```diff
@@ -183,23 +183,30 @@ def fetch_tensor_proxy(tracer):
     return lambda t: get_proxy_slot(t, tracer, t)
 
 
-def proxy_call(proxy_mode, func_overload, args, kwargs=None):
+def proxy_call(proxy_mode, func, args, kwargs=None):
     if kwargs is None:
         kwargs = {}
 
-    func = func_overload.overloadpacket
-    if func_overload in CURRENT_DECOMPOSITION_TABLE:
+    def can_handle_tensor(x):
+        return type(x) in HANDLED_TYPES or has_proxy_slot(x, proxy_mode.tracer)
+
+    # If there are any tensor subclasses, we need to handle those tensor subclasses first
+    # TODO: we could use types to test this
+    if not pytree.tree_all_only(torch.Tensor, can_handle_tensor, (args, kwargs)):
```

Comment:
The old handling this was wrong - it just happened to work for our tests.

We used to only check this in the "no proxy tensors section", but that doesn't handle cases like `add(ProxyTensor, FakeTensor)`.